### PR TITLE
feat(log streaming): add ability to stream step and service logs

### PR DESCRIPTION
**Pull request by KellyMerrick**
For https://github.com/go-vela/community/issues/366

### CODE REVIEW

**Review by wass3r — APPROVED** [review]

File: router/service.go [hunk 1]
```diff
@@ -39,6 +40,8 @@ func ServiceHandlers(base *gin.RouterGroup) {
 			service.PUT("", perm.MustPlatformAdmin(), middleware.Payload(), api.UpdateService)
 			service.DELETE("", perm.MustPlatformAdmin(), api.DeleteService)
 
+			service.POST("/stream", api.PostServiceStream)
```
Comment (jbrockopp) [inline router/service.go:43]: I wonder if we should explore locking down this endpoint behind some permission layer?

Thinking for the other endpoints we call `perm.MustPlatformAdmin()` so perhaps we adopt that?

```suggestion
			service.POST("/stream", perm.MustPlatformAdmin(), api.PostServiceStream)
```

File: router/step.go [hunk 2]
```diff
@@ -39,6 +40,8 @@ func StepHandlers(base *gin.RouterGroup) {
 			step.PUT("", perm.MustPlatformAdmin(), middleware.Payload(), api.UpdateStep)
 			step.DELETE("", perm.MustPlatformAdmin(), api.DeleteStep)
 
+			step.POST("/stream", api.PostStepStream)
```
Comment (jbrockopp) [inline router/step.go:43]: I wonder if we should explore locking down this endpoint behind some permission layer?

Thinking for the other endpoints we call `perm.MustPlatformAdmin()` so perhaps we adopt that?

```suggestion
			step.POST("/stream", perm.MustPlatformAdmin(), api.PostStepStream)
```

**Review by jbrockopp — CHANGES_REQUESTED** [review]

File: router/service.go (hunk 1 above)
Comment (wass3r) [reply router/service.go]: 👍🏼

File: router/step.go (hunk 2 above)
Comment (wass3r) [reply router/step.go]: 👍🏼

**Review by wass3r — APPROVED** [review]

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**